### PR TITLE
Revisited: feat!: Update output path writing

### PR DESCRIPTION
**Pull request by Mic92**
https://github.com/nix-community/buildbot-nix/pull/243

### CODE REVIEW

File: buildbot_nix/__init__.py [hunk 1]
```diff
@@ -784,16 +784,55 @@ def __init__(self, project: GitProject, path: Path, **kwargs: Any) -> None:
...
-        (self.path / attr).write_text(out_path)
+
+        if not out_path:  # if, e.g., the build fails and doesn't produce an output
+            return util.SKIPPED
+
+        project_name = Path(props.getProperty("projectname"))
+
+        target = Path(props.getProperty("branch"))
+
+        attr = Path(props.getProperty("attr"))
```
Comment (zowoq) [inline buildbot_nix/__init__.py:823]: Should this stay the same as it was a few lines above?
```
attr = Path(props.getProperty("attr")).name
``

Comment (Mic92) [reply buildbot_nix/__init__.py:823]: I have a look.

Comment (Mic92) [reply buildbot_nix/__init__.py:823]: I thought about it a bit, and normalizing attributes, could lead to collisions, no?

Comment (Mic92) [reply buildbot_nix/__init__.py:823]: i.e. if someone does something like this:

```
{
  "foo" = ...;
  "foo/bar" = ...;
}

Comment (Mic92) [reply buildbot_nix/__init__.py:823]: Mhm. Also thinkinng about it this specific case would create collisions anyway.

**Pull request merged**